### PR TITLE
 Support for modifying instructions (labels especially) (#2)

### PR DESCRIPTION
**Pull request by jpopelka**

### CODE REVIEW

File: dockerfile_parse/parser.py [hunk 1]
```diff
@@ -236,3 +264,73 @@ def labels(self):
                         labels[key_val[0]] = key_val[1] if len(key_val) > 1 else ''
                         logger.debug("new label %s=%s", repr(key_val[0]), repr(labels[key_val[0]]))
         return labels
+
+    def modify_instruction_label(self, label_key, label_value):
+        """
+        Rewrite a local Dockerfile with an incremented Release label
```
Comment (twaugh) [inline dockerfile_parse/parser.py]: The doc text should probably be more generic. :-)

File: dockerfile_parse/parser.py [hunk 2]
```diff
@@ -236,3 +264,73 @@ def labels(self):
                         labels[key_val[0]] = key_val[1] if len(key_val) > 1 else ''
                         logger.debug("new label %s=%s", repr(key_val[0]), repr(labels[key_val[0]]))
         return labels
+
+    def modify_instruction_label(self, label_key, label_value):
```
Comment (twaugh) [inline dockerfile_parse/parser.py]: I guess a more Pythonic way of doing it would be to use setters for the properties, but would that be really hard?

Comment (jpopelka) [inline dockerfile_parse/parser.py]: Is the additional commit what you've had in mind ?

**Comment by jpopelka** [comment]
I've added real labels setter, so you have 2 options now:
  * df.labels = {'name':'value'}
- the real setter, this *sets* (i.e. remove old ones) LABELs to the specified ones
  * df.change_labels({'name':'value'})
- this only changes value of existing LABELs

I haven't managed to make the `x.labels['name'] = 'value'` work without breaking other things :-(

**Comment by twaugh** [comment]
OK, never mind about setting `x.labels['name']` directly.

For `change_labels()` maybe we should raise an exception if we don't find the LABEL key, rather than failing an assertion.

I still think it would be nice for us to be able to make minimal changes to LABELs even when setting them, like in the tests I put here: https://github.com/jpopelka/dockerfile-parse/compare/modify-labels...twaugh:modify-labels-tests

It shouldn't be hard: getting two sets of LABEL keys, ones we already have and ones we need to end up with, we can easily find out which we need to add/delete/modify, with `set() - set()`-type calculations.

**Comment by jpopelka** [comment]
> For change_labels() maybe we should raise an exception if we don't find the LABEL key, rather than failing an assertion.

already changed to KeyError

> I still think it would be nice for us to be able to make minimal changes to LABELs even when setting them

done


> It shouldn't be hard ...

grrr, I don't want to see quotations for a veeery long time ;-)

**Comment by twaugh** [comment]
This looks really nice!

**Comment by jpopelka** [comment]
Could you check the new commit please ?
Thanks, I really appreciate all your remarks.

**Comment by jpopelka** [comment]
OK, I think I've lost this battle with quotes :-(
With the last commit when we change value of label which is on one line with other labels, those other labels will be in quotes even they weren't previously (which is not wrong per se).

You can revert this 'revert' commit if you want to try, but I'm really at my wit's end with this.
I don't want to spend more time with this as I think this state is good enough.

**Comment by twaugh** [comment]
LGTM!

**Comment by jpopelka** [comment]
Thank you Tim !

**Pull request merged**